### PR TITLE
Refactor(runtime-rs): add conditional compile for virt-sandbox persist

### PR DESCRIPTION
**Pull request by Tim-0731-Hzt**
code refactoring, add conditional compile for virt-sandbox persist

Fixes: #5706
Signed-off-by: Zhongtao Hu <zhongtaohu.tim@linux.alibaba.com>

### CODE REVIEW

**Review by studychao — COMMENTED** [review]
I'm good with the commit itself.
But I'm curious about why changing to conditional compilation when deciding sandbox type instead of just using `sandbox_state.sandbox_type`

**Comment by Tim-0731-Hzt** [comment]
> I'm good with the commit itself. But I'm curious about why changing to conditional compilation when deciding sandbox type instead of just using `sandbox_state.sandbox_type`

For other sandbox types like Linux and wasm, the function will not be called. So, I think it's better to conditionally compile it

File: src/runtime-rs/crates/runtimes/src/manager.rs [hunk 1]
```diff
@@ -153,26 +154,23 @@ impl RuntimeHandlerManager {
...
+        #[cfg(feature = "virt")]
+        {
+            let sandbox = VirtSandbox::restore(sandbox_args, sandbox_state)
+                .await
+                .context("failed to restore the sandbox")?;
+            sandbox
+                .cleanup(&inner.id)
+                .await
+                .context("failed to cleanup the resource")?;
+        }
```
Comment (bergwolf) [inline src/runtime-rs/crates/runtimes/src/manager.rs:173]: What if we enable all `linux`, `wasm` and `virt` features, and let upper layer decide which type of sandbox to create?

Comment (studychao) [reply src/runtime-rs/crates/runtimes/src/manager.rs:173]: +1

**Comment by Ji-Xinyou** [comment]
It would be more clear to change commit title to `refactor(runtime-rs): add conditional compile for virt-sandbox persist`

File: src/runtime-rs/crates/runtimes/src/manager.rs (hunk 1 above)
Comment (Tim-0731-Hzt) [reply src/runtime-rs/crates/runtimes/src/manager.rs:181]: fixed

**Review by quanweiZhou — APPROVED** [review]
LGTM

**Comment by studychao** [comment]
@Tim-0731-Hzt There is one ubuntu CI not passing, could you take a look?

**Pull request merged**